parachain/runtime/test-common/src/lib.rs

```diff
@@ -154,6 +166,29 @@ pub fn send_transfer_token_message_success<Runtime, XcmConfig>(
 				e,
 				snowbridge_pallet_outbound_queue::Event::MessageQueued { .. }
 			)));
+
+			let block_number = RuntimeHelper::<Runtime, AllPalletsWithoutSystem>::block_number();
+			let next_block_number = block_number.saturating_add(U256::from(1));
+
+			let included_head = RuntimeHelper::<Runtime, AllPalletsWithoutSystem>::run_to_block(
+				next_block_number.as_u32(),
+				AccountId::from(Alice).into(),
+			);
+
+			let origin: ParaId = (assethub_parachain_id as u32).into();
+			let channel_id: ChannelId = origin.into();
+
+			let nonce = snowbridge_pallet_outbound_queue::Nonce::<Runtime>::try_get(channel_id);
+			assert_ok!(nonce);
+			assert_eq!(nonce.unwrap(), 1);
+
+			let digest = included_head.digest();
+
+			//let digest = frame_system::Pallet::<Runtime>::digest();
+			let digest_items = digest.logs();
+			assert!(digest_items.len() == 1 && digest_items[0].as_other().is_some());
+
+			//assert_eq!(Messages::<Test>::decode_len(), Some(4));
```

Comment:
Seems it's because the [run_to_block](https://github.com/Snowfork/polkadot-sdk/blob/2afac316933a3b031c979af9cb8d1e36f03eccb8/cumulus/parachains/runtimes/test-utils/src/lib.rs#L287-L296) we run in this test case does not include the finalized phase.

I'll check if possible to make some fixes here.

Reply:
https://github.com/Snowfork/snowbridge/pull/1126

@claravanstaden I've made this PR to resolve the issue and the test passed. The bad thing it requires more changes in `polkadot-sdk` repo to include the finalized phase. Though I think it is general enough but not sure worth introducing at this moment. 

Another option is to limit the scope of the change and maybe add another `run_to_block` function in our test case to only finalize the pallets as required. Something like we do in the unit test https://github.com/Snowfork/snowbridge/blob/3b17d2a89d23ea1de5889418acae7683ecd49604/parachain/pallets/outbound-queue/src/mock.rs#L122-L136

I'll leave it up to you.

Reply:
Hey @yrong! Thanks for the effort you put in here, I appreciate it. Let me try the local "run_to_end_of_next_block" in our test case first, as I guess that will be the least disruptive.